src/DFA/Common.fs

```diff
@@ -0,0 +1,35 @@
+module PeanutProver.DFA.Common
+
+open System.Collections.Generic
+
+open Ast.Common
+open PeanutProver.Automata
+
+let renumber_transitions
+    (old_transitions: Dictionary<State, Dictionary<bit list, State>>)
+    total_variables
+    automation_indices
+    =
+    let rec selfProduct sequence number =
+        match number with
+        | 0 -> Seq.singleton []
```

Comment:
Use `if` instead. It more [idiomatic](https://stackoverflow.com/questions/7536633/ocaml-pattern-matching-vs-if-else-statements) when no destruction needed